Re-add relationship constraint documentation (#490)
neo4j/docs-cypher: modules/ROOT/pages/constraints/examples.adoc

Added 1 constraint.
----

[NOTE]
====

> The detailed statistics view would currently say `Unique constraints added: 1`, but will be updated to say `Node property uniqueness constraints added: 1` in Neo4j version 6.0.

Reviewer: ```suggestion
The detailed statistics view currently says: `Unique constraints added: 1`. 
It will be updated to say `Node property uniqueness constraints added: 1` in Neo4j version 6.0.
```
Same suggestion for the identical notes below (reason is that "would currently say" could imply that it currently does not actually say it).

Author: I had it cause we no longer shows the message in the result (which we had previously) but I see your point

.+CREATE RELATIONSHIP+
======

> Create a `SEQUEL_OF` relationship with values for properties `order` and `seriesTitle` that is not already in the database.

Reviewer: ```suggestion
Create a `SEQUEL_OF` relationship with values for properties `order` and `seriesTitle` that are not already in the database.
```

.+CREATE RELATIONSHIP+
======

> Create a `SEQUEL_OF` relationship with values for properties `order` and `seriesTitle` that is already used in the database.

Reviewer: ```suggestion
Create a `SEQUEL_OF` relationship with values for properties `order` and `seriesTitle` that are already used in the database.
```

.+CREATE CONSTRAINT+
======

> Create a property uniqueness constraint on the property `seriesTitle` on relationships with the `SEQUEL_OF` relationship type when there are two relationships with the same `seriesTitle`.

Reviewer: ```suggestion
Create a property uniqueness constraint on the property `seriesTitle` on relationships with the `SEQUEL_OF` relationship type, when two relationships with the same `seriesTitle` already exist.
```

.Error message
[source, error]
----
Unable to create Constraint( name='series_title', type='RELATIONSHIP UNIQUENESS', schema=()-[:SEQUEL_OF {seriesTitle}]-() ):
Both Relationship(0) and Relationship(1) have the type `SEQUEL_OF` and property `seriesTitle` = 'Chronicles of Ancient Darkness'
----

======

> The constraint creation fails on the first offending relationships that are found.

Reviewer 2: ```suggestion
The constraint creation will fail on the first offending relationship that is found.
```

Author: ```suggestion
The constraint creation will fail on the first offending relationships that are found.
```
the issue is _two_ relationships having the same `seriesTitle` so keeping the plural might be nice? (would also need updating for the node versions if we want to update the comment)

Reviewer: How about "the constraint creation fails on the first offending relationship that is found"?

Author: The error reports that `relationship(0)` and `relationship(1)` has the same value for the property so to me it feels odd to say `offending relationship` in singular...

Reviewer: Ok, I see your point - had not looked properly at the error message again. From my POV, I think your phrasing makes sense, then, and this PR is good to be merged.

Reviewer 2: Ah, right. Agreed, this makes sense to me: 

`The constraint creation will fail on the first offending relationships that are found.`

.+CREATE CONSTRAINT+
======

> Trying to create a relationship key constraint on the property `level` on relationships with the `KNOWS` relationship type will fail when two relationships with identical `level` already exists in the database.

Reviewer: ```suggestion
Trying to create a relationship key constraint on the property `level` on relationships with the `KNOWS` relationship type will fail when two relationships with identical `level` property values already exist in the database.
```

.Error message
[source, error]
----
Unable to create Constraint( name='knows_level', type='RELATIONSHIP KEY', schema=()-[:KNOWS {level}]-() ):
Both Relationship(0) and Relationship(1) have the type `KNOWS` and property `level` = 10
----

======

> The constraint creation fails on the first offending relationships that are found.

Reviewer 2: ```suggestion
The constraint creation will fail on the first offending relationship that is found.
```

Reviewer: Alternative: "The constraint creation fails on the first offending relationship that is found"

│13  │"sequels"                     │"RELATIONSHIP_UNIQUENESS"        │"RELATIONSHIP"│["SEQUEL_OF"]  │["order","seriesTitle"]         │"sequels"                     │
├────┼──────────────────────────────┼─────────────────────────────────┼──────────────┼───────────────┼────────────────────────────────┼──────────────────────────────┤
│21  │"wrote_locations"             │"RELATIONSHIP_PROPERTY_EXISTENCE"│"RELATIONSHIP"│["WROTE"]      │["location"]                    │null                          │
├────┼──────────────────────────────┼─────────────────────────────────┼──────────────┼───────────────┼────────────────────────────────┼──────────────────────────────┤
│20  │"wrote_year"                  │"RELATIONSHIP_PROPERTY_EXISTENCE"│"RELATIONSHIP"│["WROTE"]      │["year"]                        │null                          │
└────┴──────────────────────────────┴─────────────────────────────────┴──────────────┴───────────────┴────────────────────────────────┴──────────────────────────────┘
15 rows
----

[NOTE]
====
The `type` column returns `UNIQUENESS` for the node property uniqueness constraint and `RELATIONSHIP_UNIQUENESS` for the relationship property uniqueness constraint.

> The `type` will be updated in Neo4j version 6.0, for node property uniqueness constraint it will be updated to `NODE_PROPERTY_UNIQUENESS` and for the relationship property uniqueness constraint to `RELATIONSHIP_PROPERTY_UNIQUENESS`.

Reviewer: ```suggestion
The `type` column will be updated in Neo4j version 6.0.
Node property uniqueness constraints will be updated to `NODE_PROPERTY_UNIQUENESS` and relationship property uniqueness constraints to `RELATIONSHIP_PROPERTY_UNIQUENESS`.
```
